public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -146,8 +145,7 @@ export class PrometheusDatasource {
 
     var allQueryPromise = _.map(queries, query => {
       if (!query.instant) {
-        let range = this.clampRange(start, end, query.step);
-        return this.performTimeSeriesQuery(query, range.start, range.end);
+        return this.performTimeSeriesQuery(query, query.start, query.end);
```

Comment:
At this point you may simply pass `query` only as parameter and retrieve `start` and `end` as `query.start` and `query.end`. Then you also won't need to create the `data` object in `performTimeSeriesQuery()`, simply pass through the `query` object as it already has all the fields it needs.

Reply:
Yeah, I had that in an earlier version. I quite like to make that argument dependency explicit in the func signature.